src/main/kotlin/no/nav/familie/ef/sak/service/steg/BeregnYtelseSteg.kt

```diff
@@ -51,19 +54,52 @@ class BeregnYtelseSteg(private val tilkjentYtelseService: TilkjentYtelseService,
             else -> emptyList()
         }
 
-        // TODO: Hent tilkjentYtelse fra forrige behandling og gjør diff med ny og ta vare på kildeBehandlingId
         tilkjentYtelseService.slettTilkjentYtelseForBehandling(behandling.id)
+
         if (beløpsperioder.isNotEmpty()) {
+            val nyeAndeler = when (behandling.type) {
+                BehandlingType.FØRSTEGANGSBEHANDLING -> beløpsperioder
+                BehandlingType.REVURDERING -> andelerForRevurdering(behandling, beløpsperioder)
+                else -> error("Steg ikke støttet for type=${behandling.type}")
+            }
+
             tilkjentYtelseService.opprettTilkjentYtelse(
                     TilkjentYtelse(
                             personident = aktivIdent,
                             behandlingId = behandling.id,
-                            andelerTilkjentYtelse = beløpsperioder
+                            andelerTilkjentYtelse = nyeAndeler
                     )
             )
+        } else if (vedtak is Innvilget) {
+            error("Innvilget vedtak må ha minimum en beløpsperiode")
         }
         vedtakService.slettVedtakHvisFinnes(behandling.id)
         vedtakService.lagreVedtak(vedtakDto = vedtak, behandlingId = behandling.id)
     }
 
+    private fun andelerForRevurdering(behandling: Behandling,
+                                      beløpsperioder: List<AndelTilkjentYtelse>): List<AndelTilkjentYtelse> {
+        val forrigeBehandlingId = behandlingService.finnSisteIverksatteBehandling(behandling.fagsakId)
```

Comment:
Jeg kjenner denne klør litt. Vi bruker den ofte for å "lenke" behandlinger. Dette vil sikkert fungere, men hva skjer om vi en vakker dag finner "feil" behandling?

Reply:
finnSisteIverksatteBehandling

Reply:
Får håpe vi har noen tester som dekker det :)

Reply:
Vet ikke om alternativet er bedre, men vi kunne jo lagret forrige behandling på behandling?

Reply:
Vi kan vel det, men vi kommer ju fortsatt bruke den samme metoden til å sette det verdiet? :P